coalib/results/result_actions/ApplyPatchAction.py

```diff
@@ -16,12 +16,15 @@ def is_applicable(result, original_file_dict, file_diff_dict):
             return False
 
         try:
-            for filename in result.diffs:
-                if filename in file_diff_dict:
-                    result.diffs[filename].__add__(
-                        file_diff_dict[filename])
+            # Needed so the addition is run for all patches -> ConflictError
+            nonempty_patches = False
+            for filename, diff in result.diffs.items():
+                if diff and (filename not in file_diff_dict or
+                             diff + file_diff_dict[filename] !=
+                             file_diff_dict[filename]):
+                    nonempty_patches = True
 
-            return True
+            return nonempty_patches
         except ConflictError:
```

Comment:
the addition still happens, `+` is an invocation o `__add__`, now we use the result and compare it with the original patch!